paparazzi/src/main/java/app/cash/paparazzi/HtmlReportWriter.kt

```diff
@@ -89,60 +89,34 @@ public class HtmlReportWriter @JvmOverloads constructor(
     fps: Int
   ): FrameHandler {
     return object : FrameHandler {
+      val snapshotDir = if (fps == -1) imagesDirectory else videosDirectory
+      val goldenDir = if (fps == -1) goldenImagesDirectory else goldenVideosDirectory
       val hashes = mutableListOf<String>()
+      val file = File(snapshotDir, snapshot.toFileName(extension = "temp.png"))
+      val writer = ApngWriter(file.path.toPath(), fps)
 
       override fun handle(image: BufferedImage) {
-        hashes += writeImage(image)
+        hashes += hash(image)
+        writer.writeImage(image)
       }
 
       override fun close() {
         if (hashes.isEmpty()) return
-
-        val shot = if (hashes.size == 1) {
-          val original = File(imagesDirectory, "${hashes[0]}.png")
-          if (isRecording) {
-            val goldenFile = File(goldenImagesDirectory, snapshot.toFileName("_", "png"))
-            original.copyTo(goldenFile, overwrite = true)
-          }
-          snapshot.copy(file = original.toJsonPath())
-        } else {
-          val hash = writeVideo(hashes, fps)
-
-          if (isRecording) {
-            for ((index, frameHash) in hashes.withIndex()) {
-              val originalFrame = File(imagesDirectory, "$frameHash.png")
-              val frameSnapshot = snapshot.copy(name = "${snapshot.name} $index")
-              val goldenFile = File(goldenImagesDirectory, frameSnapshot.toFileName("_", "png"))
-              if (!goldenFile.exists()) {
-                originalFrame.copyTo(goldenFile)
-              }
-            }
-          }
-          val original = File(videosDirectory, "$hash.mov")
-          if (isRecording) {
-            val goldenFile = File(goldenVideosDirectory, snapshot.toFileName("_", "mov"))
-            if (!goldenFile.exists()) {
-              original.copyTo(goldenFile)
-            }
-          }
-          snapshot.copy(file = original.toJsonPath())
+        writer.close()
+        val hashedFile = File(snapshotDir, "${hash(hashes)}.png")
+        file.renameTo(hashedFile)
+        file.delete()
```

Comment:
since `file` is already managed by writer, should this renameTo/delete logic be moved into the `close` implementation?  would clean this up a little more, but can also see not wanting to bake that behavior into the ApngWriter; just a thought.

Reply:
I don't think so, this is a unique to HtmlReportWriter not knowing the file name until the last image has been hashed.